src/components/Switcher.vue

```diff
@@ -0,0 +1,90 @@
+<script lang="ts">
+import { defineComponent, computed, PropType } from 'vue'
+import { RadioGroup, RadioGroupLabel, RadioGroupOption } from '@headlessui/vue'
+import AIcon from './Icon.vue'
+
+export type SwitcherOption = {
+  label: string
+  value: string
+  disabled?: boolean
+  icon?: string
+}
+
+export default defineComponent({
+  name: 'ASwitcher',
+  components: {
+    RadioGroup,
+    RadioGroupLabel,
+    RadioGroupOption,
+    AIcon
+  },
+  props: {
+    modelValue: {
+      type: String,
+      required: true
+    },
+    options: {
+      type: Array as PropType<SwitcherOption[]>,
+      required: true
+    },
+    label: {
+      type: String,
+      default: ''
+    },
+    raised: {
+      type: Boolean,
+      default: false
+    },
+    disabled: {
+      type: Boolean,
+      default: false
+    }
+  },
+  emits: ['update:modelValue'],
+  setup(props, { emit }) {
+    const model = computed({
+      get: () => {
+        return props.modelValue
+      },
+      set: value => {
+        emit('update:modelValue', value)
+      }
+    })
+    return {
+      model
+    }
+  }
+})
+</script>
+<template>
+  <RadioGroup
+    v-model="model"
+    class="p-[2px] rounded bg-whisper flex max-w-fit gap-1"
+    :class="{ 'shadow-sm': raised }"
+    :disabled="disabled">
+    <RadioGroupLabel class="sr-only">{{ label }}</RadioGroupLabel>
+    <RadioGroupOption
+      v-for="option in options"
+      :key="option.value"
+      v-slot="{ checked }"
+      :value="option.value"
+      :disabled="option.disabled || disabled"
+      class="focus-visible:focus-outline flex h-8 rounded transition-colors duration-500 ease-out">
+      <div
+        class="flex items-center px-3 py-1 rounded enabled:active:bg-zurich48 enabled:active:text-mediumblue enabled:cursor-pointer"
+        :class="{
+          'text-mediumblue bg-zurich': checked,
+          'hover:bg-gray': !checked && !option.disabled && !disabled,
+          'opacity-40 cursor-not-allowed': option.disabled || disabled
+        }"
+        :aria-label="option.icon ? option.label : undefined">
+        <slot :name="option.value">
```

Comment:
Using the value as the slot name is clever! Are there any caveats with dynamic options/slots? We have at least one place where options would be added dynamically to the switcher:

<img width="178" alt="Screenshot 2023-11-14 at 16 41 57" src="https://github.com/archilogic-com/honeycomb/assets/1694288/8ba586ba-5822-49e6-a004-b28a7529b5ff">

<img width="161" alt="Screenshot 2023-11-14 at 16 42 02" src="https://github.com/archilogic-com/honeycomb/assets/1694288/21278df9-9c05-49ae-a2d4-38c27f19046b">

... this case wouldn't even need slots, but still worth making note of it in the docs if it's not possible to dynamically add them

Reply:
Good idea!
there are some constraints mentioned in the vue docs, I'll add these links https://vuejs.org/guide/components/slots.html#dynamic-slot-names
https://vuejs.org/guide/essentials/template-syntax.html#directives